File: src/etch/bindings/csharp/transport/SourceHandler.cs
using System;
using System.Text;
using System.Collections.Generic;
using Etch.Transport;

namespace Etch.Transport
{
    /// <summary>
    /// SourceHandler receives notification of source events.
    /// </summary>
    /// <typeparam name="S">event originator</typeparam>
    public interface SourceHandler<S> : Session where S : Source
    {
        /// <summary>
        /// 
        /// </summary>
        /// <returns>the source</returns>
        S GetSource();

        /// <summary>
        /// Sets the source
        /// </summary>
        /// <param name="src"></param>
        void SetSource( S src );
    }
}

File: src/metreos/core/io/SourceHandler.java
// $Id$
//
// Created by wert on Apr 8, 2007.
//
// Copyright (c) 2007 Cisco Systems, Inc. All rights reserved.

package metreos.core.io;

import etch.bindings.java.transport.Session;

/**
 * SourceHandler receives notification of source events.
 * @param <S> event originator
 */
public interface SourceHandler<S extends Source> extends Session
{
	/**
	 * @return the source.
	 */
	public S getSource();
	
	/**
	 * Sets the source.
	 * @param src
	 */
	public void setSource( S src );
}
